### ISSUE TITLE
`hasOne` ignores `mapsTo`

### ISSUE THREAD

**Issue by ierceg** [dresende/node-orm2#638]
I have a model defined with `mapsTo` for each property (table is snake_case while model is camelCase). If I use `hasOne` to associate it to another model, the values of association fields are always null. I have tried the following alternatives:
1. Defining association field name as it appears in the model. This leads to an exception due to ORM  trying to insert a row with camelCase field.
2. Doing #1 but also changing the code in `Utilities.js` from:
```
	new_obj[obj].mapsTo = obj;
```
to
```
	new_obj[obj].mapsTo = new_obj[obj].mapsTo || obj;
```
This 2nd approach works in my tests but it broke other unit tests.

Before I continue I would like to know if I'm on the right track here. Are associations defined per model property name (camelCase in my case) or per table column name (snake_case in my case)?

**Comment by dxg** [comment]
Not sure I understand the question, but maybe neither?
I don't think I ever tested `mapsTo` with associations. The functions in `utilities.js` are really confusing to work with.

**Comment by dxg** [comment]
I've made some changes to master. They won't necessarily fix your problem, but [they include tests](https://github.com/dresende/node-orm2/blob/master/test/integration/association-hasone.js#L385).

Can you please run your code against master, and either add a failing test case, give me a code sample, or tell me how to make it fail?

**Comment by ierceg** [comment]
@dxg Seems to be fixed - I tried two different tests and both passed. Thanks for this!

**Comment by ierceg** [comment]
@dxg I misspoke - one test is still failing, I just didn't have it correctly configured. I'll add the failing test case.

**Comment by ierceg** [comment]
@dxg I rechecked, again, and it works alright. It's the mapped name that has to be specified in the `hasOne` (I had column name and then it reverts to the behavior of using NULL for the value of the relationship field)

**Comment by ierceg** [comment]
@dxg When do you expect to bump the package version?

**Comment by dxg** [comment]
Just published 2.1.25

**Comment by ierceg** [comment]
@dxg Thanks!
